subsys/net/ip/net_if.c

```diff
@@ -3990,7 +3990,12 @@ int net_if_up(struct net_if *iface)
 	/* In many places it's assumed that link address was set with
 	 * net_if_set_link_addr(). Better check that now.
 	 */
-	NET_ASSERT(net_if_get_link_addr(iface)->addr != NULL);
+	if (IS_ENABLED(CONFIG_NET_SOCKETS_CAN) &&
+	    (net_if_l2(iface) == &NET_L2_GET_NAME(CANBUS_RAW)))	{
```

Comment:
@alexanderwachter There's `CANBUS_RAW_L2` and `CANBUS_L2` - should the LL address check be skipped for both or is the current solution enough?

Reply:
RAW only. The other is 6loCAN, which is IPv6 over CAN. It has a 14bit address.